modules/spring-config/src/main/java/org/mule/runtime/config/api/dsl/model/metadata/ComponentAstBasedMetadataCacheIdGenerator.java

```diff
@@ -185,6 +186,11 @@ public ComponentAstBasedMetadataCacheIdGenerator(ComponentLocator<ComponentAst>
                                       .orElse(format("(%s):(%s)", getSourceElementName(component), "Unknown Type"))));
   }
 
+  private Optional<MetadataCacheId> resolveDslTagNamespace(ComponentAst elementModel) {
+    String namespace = elementModel.getIdentifier().getNamespace();
```

Comment:
I think the identifier can be null, we should consider that

Reply:
org.mule.runtime.ast.api.util.BaseComponentAstDecorator#toString it is not checking for null. It should always have an identifier, if it is null I would say there is a bug in AST.

Reply:
Confirmed by Rodro, it should not be null.